<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1050</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>校园活动报名与签到</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="topBarLayout">
      <item>
       <widget class="QLabel" name="userInfoLabel">
        <property name="text">
         <string>当前用户: </string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="topSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="logoutButton">
        <property name="text">
         <string>退出登录</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::TabPosition::North</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tabDashboard">
       <attribute name="title">
        <string>首页</string>
       </attribute>
       <layout class="QVBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QGroupBox" name="annGroup">
          <property name="title">
           <string>公告</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QListWidget" name="announcementList"/>
           </item>
           <item>
            <widget class="QPushButton" name="refreshAnnouncementsButton">
             <property name="text">
              <string>刷新公告</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="upcomingGroup">
          <property name="title">
           <string>近期活动</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTableView" name="upcomingTable"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabActivities">
       <attribute name="title">
        <string>活动管理</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="filterLayout">
          <item>
           <widget class="QComboBox" name="categoryFilter"/>
          </item>
          <item>
           <widget class="QComboBox" name="statusFilter"/>
          </item>
          <item>
           <widget class="QLineEdit" name="keywordEdit">
            <property name="placeholderText">
             <string>关键字搜索</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="newActivityButton">
            <property name="text">
             <string>发布新活动</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableView" name="activityTable"/>
        </item>
        <item>
         <widget class="QGroupBox" name="formGroup">
          <property name="title">
           <string>活动详情 / 编辑</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="labelTitle">
             <property name="text">
              <string>标题</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="titleEdit"/>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="labelCategory">
             <property name="text">
              <string>类别</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QComboBox" name="categoryEdit"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelLocation">
             <property name="text">
              <string>地点</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="locationEdit"/>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="labelCapacity">
             <property name="text">
              <string>容量</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QSpinBox" name="capacitySpin">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="labelStart">
             <property name="text">
              <string>开始时间</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDateTimeEdit" name="startEdit">
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="labelEnd">
             <property name="text">
              <string>结束时间</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QDateTimeEdit" name="endEdit">
             <property name="calendarPopup">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="labelStatus">
             <property name="text">
              <string>状态</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLineEdit" name="statusEdit">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QPushButton" name="submitActivityButton">
             <property name="text">
              <string>提交/更新</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QPushButton" name="approveButton">
             <property name="text">
              <string>审核通过</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QPushButton" name="rejectButton">
             <property name="text">
              <string>驳回/取消</string>
             </property>
            </widget>
           </item>
           <item row="4" column="3">
            <widget class="QPushButton" name="deleteButton">
             <property name="text">
              <string>删除</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabEnrollment">
       <attribute name="title">
        <string>学生报名</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="labelActFilter">
            <property name="text">
             <string>可报名活动</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="checkConflictButton">
            <property name="text">
             <string>检查冲突</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableView" name="enrollmentActivityTable"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="enrollButton">
            <property name="text">
             <string>报名</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="cancelEnrollButton">
            <property name="text">
             <string>取消报名</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="waitlistButton">
            <property name="text">
             <string>加入候补</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="exportMyEnrollButton">
            <property name="text">
             <string>导出我的报名 CSV</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="waitlistGroup">
          <property name="title">
           <string>我的报名 / 候补</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QTableView" name="waitlistTable"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabReports">
       <attribute name="title">
        <string>统计与导出</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="statsGroup">
          <property name="title">
           <string>概览</string>
          </property>
          <layout class="QGridLayout" name="gridLayoutStats">
           <item row="0" column="0">
            <widget class="QLabel" name="labelTotalAct">
             <property name="text">
              <string>活动总数: 0</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="labelTotalEnroll">
             <property name="text">
              <string>报名总数: 0</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelApproved">
             <property name="text">
              <string>已审核: 0</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="labelPending">
             <property name="text">
              <string>待审核: 0</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QPushButton" name="runReportButton">
            <property name="text">
             <string>生成报表 (后台)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="exportCsvButton">
            <property name="text">
             <string>导出报名列表 CSV</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="reportStatusLabel">
            <property name="text">
             <string>状态: 空闲</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableView" name="reportPreviewTable"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
